src/components/molecules/Visualizer/Layers/store.ts

```diff
@@ -68,6 +73,72 @@ export class LayerStore {
     return l;
   }
 
+  #usedIds: string[] = [];
+  #newId = () => {
+    const genResId = () =>
+      "_xxxxxxxxxxxxxxxxxxxxxxxxx".replace(/[x]/g, function () {
+        return ((Math.random() * 16) | 0).toString(16);
+      });
+    let id;
+    do {
+      id = genResId();
+    } while (this.#usedIds.includes(id));
+    this.#usedIds.push(id);
+    return id;
+  };
+
+  #insertLayer = (layerData: AddedLayerData, save = false) => {
+    const tar = layerData.parentId ? this.#map.get(layerData.parentId) : this.#root;
+    if (!tar?.children) return;
+    tar.children.push(layerData.layer);
+    const flattenedLayers = flattenLayers([layerData.layer]);
+    flattenedLayers.forEach(l => {
+      this.#map.set(l.id, l);
+      this.#pmap.set(l.id, this.#pluginLayer(l));
+    });
+    save && !tar.creator && this.#addedLayersData.push(layerData);
+    return layerData.layer.id;
+  };
+
+  #processLayer = (layer: Omit<Layer, "id">): Layer => {
+    const { children, ...self } = layer;
+    self.infobox?.blocks &&
+      self.infobox.blocks.forEach(b => {
+        b.id = this.#newId();
+      });
+    return {
+      ...self,
+      id: this.#newId(),
+      pluginId: "reearth",
+      creator: "pluginAPI",
```

Comment:
Instead of a fixed string, creator should be the ID of the plugin. Then we can implement a method to bulk delete layers added by a particular plugin. This is useful for cleanup when a plugin is destroyed.